packages/web-client/app/utils/typed-channel.ts

```diff
@@ -0,0 +1,31 @@
+export class TypedChannel<MessageType = any> {
+  channelName: string;
+  private broadcastChannel: BroadcastChannel;
+
+  constructor(channelName: string) {
+    this.channelName = channelName;
+    this.broadcastChannel = new BroadcastChannel(channelName);
+  }
+
+  addEventListener(
+    event: keyof BroadcastChannelEventMap,
+    callback: (ev: MessageEvent<MessageType>) => any
+  ) {
+    this.broadcastChannel.addEventListener(event, callback);
+  }
+
+  removeEventListener(
+    event: keyof BroadcastChannelEventMap,
+    callback: (ev: MessageEvent<MessageType>) => any
+  ) {
+    this.broadcastChannel.removeEventListener(event, callback);
+  }
+
+  close() {
+    this.broadcastChannel.close();
+  }
+
+  postMessage(message: MessageType) {
+    this.broadcastChannel.postMessage(message);
+  }
+}
```

Comment:
nice, I hope this wrapper ends up being helpful! 🤩

Reply:
For now it is only a subset of broadcast channel's functionality, we can add on as needed!